src/main/java/org/cryptomator/linux/quickaccess/DolphinPlaces.java

```diff
@@ -0,0 +1,161 @@
+import java.io.StringReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * Implemenation of the {@link QuickAccessService} for KDE desktop environments using Dolphin file browser.
+ */
+@DisplayName("KDE Dolphin Places")
+@OperatingSystem(OperatingSystem.Value.LINUX)
+@Priority(90)
+public class DolphinPlaces implements QuickAccessService {
+
+	private static final int MAX_FILE_SIZE = 1 << 15; //xml is quite verbose
+	private static final Path PLACES_FILE = Path.of(System.getProperty("user.home"), ".local/share/user-places.xbel");
+	private static final Path TMP_FILE = PLACES_FILE.resolveSibling("user-places.xbel.cryptomator.tmp");
+	private static final Lock MODIFY_LOCK = new ReentrantReadWriteLock().writeLock();
+	private static final String ENTRY_TEMPLATE = """
+			<bookmark href="%s">
+			 <title>%s</title>
+			 <info>
+			  <metadata owner="http://freedesktop.org">
+			   <bookmark:icon name="drive-harddisk"/>
+			  </metadata>
+			  <metadata owner="https://cryptomator.org">
+			  	<id>%s</id>
+			  </metadata>
+			 </info>
+			</bookmark>""";
+
+
+	private static final Validator xmlValidator;
+
+	static {
+		SchemaFactory factory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+		try (var schemaDefinition = DolphinPlaces.class.getResourceAsStream("/xbel-1.0.xsd")) {
+			Source schemaFile = new StreamSource(schemaDefinition);
+			xmlValidator = factory.newSchema(schemaFile).newValidator();
+		} catch (IOException | SAXException e) {
+			throw new IllegalStateException("Failed to load included XBEL schema definition file.", e);
+		}
+	}
+
+
+	@Override
+	public QuickAccessService.QuickAccessEntry add(Path target, String displayName) throws QuickAccessServiceException {
+		String id = UUID.randomUUID().toString();
+		try {
+			MODIFY_LOCK.lock();
+			if (Files.size(PLACES_FILE) > MAX_FILE_SIZE) {
+				throw new IOException("File %s exceeds size of %d bytes".formatted(PLACES_FILE, MAX_FILE_SIZE));
+			}
+			var placesContent = Files.readString(PLACES_FILE);
+			//validate
+			xmlValidator.validate(new StreamSource(new StringReader(placesContent)));
+			// modify
+			int insertIndex = placesContent.lastIndexOf("</xbel>"); //cannot be -1 due to validation
+			try (var writer = Files.newBufferedWriter(TMP_FILE, StandardCharsets.UTF_8, StandardOpenOption.WRITE, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)) {
+				writer.write(placesContent, 0, insertIndex);
+				writer.newLine();
+				writer.write(ENTRY_TEMPLATE.formatted(target.toUri(), displayName, id).indent(1));
+				writer.newLine();
+				writer.write(placesContent, insertIndex, placesContent.length() - insertIndex);
+			}
+			// save
```

Comment:
I'd suggest validating our modified file a second time before moving it.

Reply:
We can ensure during development, that we don't create an invalid XML document. 😉 Especially writing is quite clear:
1. find end tag
2. write everything before end tag
3. write our entry
4. write end tag and everything after